File: Array/1431.Kids_With_the_Greatest_Number_of_Candies/C++/1431.Kids_With_the_Greatest_Number_of_Candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int> & candies, int extraCandies) 
    {
        //=======================================//
        // Input type:                           //
        //   - vector<int> ref. (candies vector) //
        //   - int (extraCandies value)          //
        // Return type:                          //
        //   - vector<bool> (result vector)      //
        //=======================================//

        /*Initialize*/
        ///// Length, Maximun of candies vector /////
        int len_candies = candies.size(), max_candies = *max_element(candies.begin(), candies.end());

        ///// Result vector /////
        vector<bool> res_vect(len_candies, false);


        /*One-pass update based loop traversal*/
        for (int candies_idx = 0; (candies_idx < len_candies); (candies_idx++)) //Whole
        {
            ///// Check if the current summary value matched conditions or not /////
            if ((candies[candies_idx] + extraCandies) >= max_candies) { res_vect[candies_idx] = true; } //Keep updating, overwriting
            else { ; }

        } //Whole

        return res_vect;
    }
};

File: Array/1431.Kids_With_the_Greatest_Number_of_Candies/Python/1431.Kids_With_the_Greatest_Number_of_Candies.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """
        #========================================#
        # One-pass update based traversal method #
        #========================================#

        ############
        #Initialize
        ##### Length, Maximun of candies array #####
        len_candies, max_candies = len(candies), max(candies)

        ##### Result array #####
        res_arry = [False for _ in range(len_candies)]


        ######################################
        #One-pass update based loop traversal
        for candies_idx in range(len_candies):

            ##### Check if the current summary value matched conditions or not #####
            if ((candies[candies_idx] + extraCandies) >= max_candies):
                res_arry[candies_idx] = True #Keep updating/overwriting
            else:
                pass

        return res_arry
